Source/SwiftLintFramework/Rules/QuickDiscouragedCallRule.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  QuickDiscouragedCallRule.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/11/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+// swiftlint:disable identifier_name
+
+public struct QuickDiscouragedCallRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "quick_discouraged_call",
+        name: "Quick Discouraged Call",
+        description: "Discouraged call inside 'describe' and/or 'context' block.",
+        kind: .style,
+        nonTriggeringExamples: QuickDiscouragedCallRuleExamples.nonTriggeringExamples,
+        triggeringExamples: QuickDiscouragedCallRuleExamples.triggeringExamples
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            fileContainsQuickSpec(file: file),
```

Comment:
@ornithocoder I've put together an alternative approach in https://github.com/realm/SwiftLint/commit/a2552056f0503f84dc74999db17e3bfcd9f6d43d. Please take a look and tell me what do you think. It should be faster since we don't need to navigate the whole structure for each violation.